integration/subxt-tests/src/lib.rs

```diff
@@ -0,0 +1,548 @@
+use contract_transcode::ContractMessageTranscoder;
+
+use node::runtime_types::pallet_contracts::wasm::Determinism;
+use pallet_contracts_primitives::{
+    ContractExecResult, ContractResult, ExecReturnValue, GetStorageResult,
+};
+use parity_scale_codec::{Decode, Encode};
+
+use sp_core::{crypto::AccountId32, hexdisplay::AsBytesRef, Bytes};
+use sp_weights::Weight;
+use subxt::{
+    blocks::ExtrinsicEvents as TxEvents,
+    ext::sp_runtime::DispatchError,
+    tx::PairSigner,
+    utils::{MultiAddress, Static},
+    Config, OnlineClient, PolkadotConfig,
+};
+
+use contract_metadata::ContractMetadata;
+use sp_keyring::AccountKeyring;
+use tokio::time::timeout;
+
+mod cases;
+
+// metadata file obtained from the latest substrate-contracts-node
+#[subxt::subxt(
+    runtime_metadata_path = "./metadata.scale",
+    substitute_type(
+        type = "sp_weights::weight_v2::Weight",
+        with = "::subxt::utils::Static<::sp_weights::Weight>"
+    )
+)]
+pub mod node {}
```

Comment:
IIRC `subxt` can download the metadata from the node, so we don't have to maintain it here (the typescript test suite doesn't need this). Could you check if it is possible?

Reply:
Yeah this is exactly what I had in mind 👍  (couldn't find it though)